http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -0,0 +1,194 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.effect.{Async, Concurrent, ContextShift, IO, Timer}
+import org.broadinstitute.dsde.workbench.google2.{Event, GoogleSubscriber}
+import fs2.{Pipe, Stream}
+import io.circe.{Decoder, DecodingFailure, Encoder}
+import org.broadinstitute.dsde.workbench.leonardo.MachineConfig
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, followupQuery, DbReference}
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterStatus, MachineType}
+import org.broadinstitute.dsde.workbench.leonardo.util.ClusterHelper
+import _root_.io.chrisdavenport.log4cats.Logger
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterStatus.Stopped
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage._
+import org.broadinstitute.dsde.workbench.model.WorkbenchException
+import cats.implicits._
+
+import scala.concurrent.ExecutionContext
+
+case class ClusterNotFoundException(
+  clusterId: Long,
+  override val message: String =
+    "Could not process ClusterTransitionFinishedMessage because it was not found in the database"
+) extends LeoException
+
+class LeoPubsubMessageSubscriber[F[_]: Async: Timer: ContextShift: Logger: Concurrent](
+  subscriber: GoogleSubscriber[IO, LeoPubsubMessage],
+  clusterHelper: ClusterHelper,
+  dbRef: DbReference[IO]
+)(implicit executionContext: ExecutionContext)
+    extends LazyLogging {
+
+  def messageResponder(message: LeoPubsubMessage): IO[Unit] = {
+    val response = message match {
+      case msg @ StopUpdateMessage(_, _) =>
+        handleStopUpdateMessage(msg)
+      case msg @ ClusterTransitionFinishedMessage(_) =>
+        handleClusterTransitionFinished(msg)
+      case _ => IO.unit
+    }
+
+    //ensure we don't crash if the handler throws an exception
+    response.onError {
+      case e => IO(logger.error(s"Unable to process a message in received from pub/sub subscription in messageResponder: ${message}, errorMessage: ${e.getMessage}", e))
```

Comment:
So it seems like errors are just logged, correct? I wonder if we should do any other action, like put the cluster in Error state, or maybe have a timeout for the entire stop-update transition. I'm not really sure yet, will think about it..

Reply:
As of right now, I think logging is the appropriate response, as the cluster won't be in a bad state and we will acknowledge the message and flush it from the system. From the users perspective, if this happens, their cluster will not go through the desired transition, but it should still display the correct state in the UI throughout. It could be confusing as their update won't finish, but it should recover naturally.